### PR TITLE
Add support for generic OTEL_PYTHON_EXCLUDED_URLS variable

### LINKED ISSUE #789: Add support for generic OTEL_PYTHON_EXCLUDED_URLS variable
**Is your feature request related to a problem?**
Right now, to exclude URLs from being traced using environment variables, the `OTEL_PYTHON_{instrumentation}_EXCLUDED_URLS` naming convention must be used.
However, considering that the inner workings of the excluded URLs are independent of the running integration, and actually related to the `ExcludedList` implementation, having support for a generic `OTEL_PYTHON_EXCLUDED_URLS` would simplify deployments of different stacks, where DevOps are in charge of creating reusable modules that centralize OpenTelemetry configuration (e.g. `OTEL_PROPAGATORS`, `OTEL_PYTHON_ID_GENERATOR`).

**Describe the solution you'd like**
Having this library read the `OTEL_PYTHON_EXCLUDED_URLS` environment variable, as a fallback when no value is found at `OTEL_PYTHON_{implementation}_EXCLUDED_URLS`.

**Describe alternatives you've considered**
As it is today, having reusable modules for Infrastructure-as-Code requires to either know which application is being deployed (Django, FastAPI, Tornado...), to set the related environment variable, or set all of them even if not required, to simplify the logic and avoid developers needing to be explicit about the framework being used.

For the solution, an alternative approach would be for `OTEL_PYTHON_EXCLUDED_URLS` to extend what's provided by `OTEL_PYTHON_{implementation}_EXCLUDED_URLS`. However, that could be a separate discussion.

### PR DESCRIPTION
**Pull request by adamantike**
# Description

Use `OTEL_PYTHON_EXCLUDED_URLS` environment variable as a fallback of `OTEL_PYTHON_{instrumentation}_EXCLUDED_URLS`.

Fixes #789

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] `tox -e test-util-http`

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py [hunk 1]
```diff
@@ -47,24 +48,28 @@ def get_traced_request_attrs(instrumentation):
     return traced_request_attrs
 
 
-def get_excluded_urls(instrumentation):
+def get_excluded_urls(instrumentation: str) -> ExcludeList:
     excluded_urls = environ.get(
-        _root.format(f"{instrumentation}_EXCLUDED_URLS"), []
+        _root.format(f"{instrumentation}_EXCLUDED_URLS"), ""
```
Comment (owais) [inline util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py:53]: If a user sets `OTEL_PYTHON_EXCLUDED_URLS=/something` and then sets `OTEL_PYTHON_DJANGO_EXLUDED_URLS=""` in order to exclude django from the global URLs, would this logic force force Django instrumentation to use global exclude list still?

Comment (adamantike) [reply util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py:53]: Yes, that will be the case. If we want `OTEL_PYTHON_DJANGO_EXCLUDED_URLS` to be able to disable excluded URLs entirely, we can move to this instead:

```python
def get_excluded_urls(instrumentation: str) -> ExcludeList:
    excluded_urls = environ.get(
        _root.format(f"{instrumentation}_EXCLUDED_URLS"),
        environ.get(_root.format("EXCLUDED_URLS"), "")
    )

    return parse_excluded_urls(excluded_urls)
```

Comment (adamantike) [reply util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py]: Change applied to always respect instrumentation-specific variable if set.

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py [hunk 2]
```diff
@@ -32,7 +32,8 @@
 
 Exclude lists
 *************
-To exclude certain URLs from being tracked, set the environment variable ``OTEL_PYTHON_DJANGO_EXCLUDED_URLS`` with comma delimited regexes representing which URLs to exclude.
+To exclude certain URLs from being tracked, set the environment variable ``OTEL_PYTHON_EXCLUDED_URLS`` or ``OTEL_PYTHON_DJANGO_EXCLUDED_URLS``
+with comma delimited regexes representing which URLs to exclude.
```
Comment (lonewolf3739) [inline instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py:36]: Maybe mention the precedence when both are set?

**Review by lonewolf3739 — APPROVED** [review]

**Review by owais — APPROVED** [review]

**Review by lzchen — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py (hunk 2 above)
Comment (adamantike) [reply instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py:36]: Would this wording make that clear, or should we go with a different one?
> To exclude certain URLs from being tracked, set the environment variable ``OTEL_PYTHON_{instrumentation}_EXCLUDED_URLS`` (or ``OTEL_PYTHON_EXCLUDED_URLS`` as fallback) with comma delimited regexes representing which URLs to exclude.

**Comment by ocelotl** [comment]
@adamantike please resolve conflicts so that we can merge :v:

**Comment by adamantike** [comment]
Rebased, and changed docs wording. Let me know if that's clear enough!

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py (hunk 2 above)
Comment (owais) [reply instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py]: @lonewolf3739 please resolve if you are satisfied with the change. thanks

**Pull request merged**